src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -183,13 +189,141 @@ private void commitEntriesToStorage(Map<WALEntry, WALEntry> entries) {
+    // so we only need to track the committed filename (values).
+    ArrayList<String> committedFilenames = new ArrayList<>();
+
+    WALEntry key = new WALEntry();
+    WALEntry value = new WALEntry();
+    boolean wasBeginMarker = false;
+
+    // The entry with the latest offsets will be in the last BEGIN-END block of the file.
+    // There may be empty BEGIN and END blocks as well.
+    while (reader.next(key, value)) {
+      String keyName = key.getName();
+      if (keyName.equals(beginMarker)) {
+        wasBeginMarker = true;
+      } else if (!keyName.equals(endMarker)) {
+        if (wasBeginMarker) {
+          // if we hit a filename entry and the previous token was a BEGIN marker
+          // we start a new BEGIN-END block
+          committedFilenames.clear();
+        }
+        committedFilenames.add(value.getName());
+        wasBeginMarker = false;
+      }
+    }
+    return committedFilenames;
+  }
+
+  /**
+   * Extract the offsets from the given filenames and find the latest offset and filepath.
+   *
+   * @param committedFileNames a list of committed filenames
+   * @return the latest offset committed along with the filepath
+   */
+  private Pair<Long, String> getLatestOffsetFromList(List<String> committedFileNames) {
+    Pair<Long, String> latestOffsetEntry = null;
+    long latestOffset = -1;
+    // Entries in the BEGIN-END block are currently not guaranteed any ordering.
+    for (String fileName : committedFileNames) {
+      long currentOffset = extractOffsetsFromFilePath(fileName);
+      if (currentOffset > latestOffset) {
+        latestOffset = currentOffset;
+        latestOffsetEntry = Pair.of(latestOffset, fileName);
+      }
+    }
+    return latestOffsetEntry;
+  }
+
+  /**
+   * Extract the file offset from the full file path.
+   *
+   * @param fullPath the full HDFS file path
+   * @return the offset or -1 if not present
+   */
+  static long extractOffsetsFromFilePath(String fullPath) {
+    try {
+      if (fullPath != null) {
+        String latestFileName = Paths.get(fullPath).getFileName().toString();
+        return FileUtils.extractOffset(latestFileName);
+      }
+    } catch (IllegalArgumentException e) {
+      log.warn("Could not extract offsets from file path: {}", fullPath);
+    }
+    return -1;
+  }
+
+  private Reader newWalFileReader() throws IOException {
+    return new Reader(conf.getHadoopConfiguration(), Reader.file(new Path(logFile)));
+  }
+
   @Override
   public void truncate() throws ConnectException {
     log.debug("Truncating WAL file: {}", logFile);
     try {
-      String oldLogFile = logFile + ".1";
-      storage.delete(oldLogFile);
-      storage.commit(logFile, oldLogFile);
+      String oldLogFile = logFile + OLD_LOG_EXTENSION;
+      if (storage.exists(oldLogFile)) {
+        storage.delete(oldLogFile);
+      }
+      if (storage.exists(logFile)) {
+        storage.commit(logFile, oldLogFile);
```

Comment:
why do we need this?

Reply:
Do you mean the `storage.commit(logFile, oldLogFile);` part? That is how the connector writes the recent `WAL` file to the truncated (old) file thereby creating the truncated file with the `.1` extension. If that commit wasn't present, the truncated `WAL`s wouldn't exist.

Reply:
sorry, i meant the `exists` calls

Reply:
I added the exist calls due to seeing any exist checks in the chain of calls for the `delete` and `commit` file system methods.

Reply:
but its been working before? interesting

Reply:
Yes it's been working before, +1 to the same effect, I'll revert this.